<h1> ACTIVITY 1 - PALMER PENGUINS DATA ANALYTICS PROJECT </h1> <hr> 

<h1>Analyst : Jaynard Diosay</h1>

In [1]:
#libraries

In [2]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

In [3]:
penguin = pd.read_csv('penguins.csv')

In [4]:
#penguin = pd.read_csv('dataset\\penguins.csv')

In [5]:
penguin

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<hr> <h1>Descriptive Analytics</h1>

In [6]:
penguin.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [7]:
penguin.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

<hr> <h1>Filtering a Column / Data </h1>

In [31]:
penguin['species']

0         Adelie
1         Adelie
2         Adelie
3         Adelie
4         Adelie
         ...    
339    Chinstrap
340    Chinstrap
341    Chinstrap
342    Chinstrap
343    Chinstrap
Name: species, Length: 344, dtype: category
Categories (3, object): ['Adelie', 'Chinstrap', 'Gentoo']

In [8]:
penguin ['species'] == 'Adelie'

0       True
1       True
2       True
3       True
4       True
       ...  
339    False
340    False
341    False
342    False
343    False
Name: species, Length: 344, dtype: bool

In [9]:
adelie_df = penguin[penguin['species'] == "Adelie"]

In [10]:
adelie_df['body_mass_g'].mean()

np.float64(3700.662251655629)

In [11]:
penguin['species'].value_counts()

penguin [(penguin['species'] == 'Adelie') & (penguin['island'] == 'Dream')]

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
30   Adelie  Dream            39.5           16.7              178.0   
31   Adelie  Dream            37.2           18.1              178.0   
32   Adelie  Dream            39.5           17.8              188.0   
33   Adelie  Dream            40.9           18.9              184.0   
34   Adelie  Dream            36.4           17.0              195.0   
35   Adelie  Dream            39.2           21.1              196.0   
36   Adelie  Dream            38.8           20.0              190.0   
37   Adelie  Dream            42.2           18.5              180.0   
38   Adelie  Dream            37.6           19.3              181.0   
39   Adelie  Dream            39.8           19.1              184.0   
40   Adelie  Dream            36.5           18.0              182.0   
41   Adelie  Dream            40.8           18.4              195.0   
42   Adelie  Dream            36.0           18.5              186.0   
43   Adelie  Dream            44.1           19.7              196.0   
44   Adelie  Dream            37.0           16.9              185.0   
45   Adelie  Dream            39.6           18.8              190.0   
46   Adelie  Dream            41.1           19.0              182.0   
47   Adelie  Dream            37.5           18.9              179.0   
48   Adelie  Dream            36.0           17.9              190.0   
49   Adelie  Dream            42.3           21.2              191.0   
84   Adelie  Dream            37.3           17.8              191.0   
85   Adelie  Dream            41.3           20.3              194.0   
86   Adelie  Dream            36.3           19.5              190.0   
87   Adelie  Dream            36.9           18.6              189.0   
88   Adelie  Dream            38.3           19.2              189.0   
89   Adelie  Dream            38.9           18.8              190.0   
90   Adelie  Dream            35.7           18.0              202.0   
91   Adelie  Dream            41.1           18.1              205.0   
92   Adelie  Dream            34.0           17.1              185.0   
93   Adelie  Dream            39.6           18.1              186.0   
94   Adelie  Dream            36.2           17.3              187.0   
95   Adelie  Dream            40.8           18.9              208.0   
96   Adelie  Dream            38.1           18.6              190.0   
97   Adelie  Dream            40.3           18.5              196.0   
98   Adelie  Dream            33.1           16.1              178.0   
99   Adelie  Dream            43.2           18.5              192.0   
132  Adelie  Dream            36.8           18.5              193.0   
133  Adelie  Dream            37.5           18.5              199.0   
134  Adelie  Dream            38.1           17.6              187.0   
135  Adelie  Dream            41.1           17.5              190.0   
136  Adelie  Dream            35.6           17.5              191.0   
137  Adelie  Dream            40.2           20.1              200.0   
138  Adelie  Dream            37.0           16.5              185.0   
139  Adelie  Dream            39.7           17.9              193.0   
140  Adelie  Dream            40.2           17.1              193.0   
141  Adelie  Dream            40.6           17.2              187.0   
142  Adelie  Dream            32.1           15.5              188.0   
143  Adelie  Dream            40.7           17.0              190.0   
144  Adelie  Dream            37.3           16.8              192.0   
145  Adelie  Dream            39.0           18.7              185.0   
146  Adelie  Dream            39.2           18.6              190.0   
147  Adelie  Dream            36.6           18.4              184.0   
148  Adelie  Dream            36.0           17.8              195.0   
149  Adelie  Dream            37.8           18.1              193.0   
150  Adelie  Dream            36.0          

In [32]:
penguin['body_mass_g'] >=4000

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340    False
341    False
342     True
343    False
Name: body_mass_g, Length: 344, dtype: bool

In [33]:
(penguin['body_mass_g'] >=4000) & (penguin['species'] == 'Adelie')

0      False
1      False
2      False
3      False
4      False
       ...  
339    False
340    False
341    False
342    False
343    False
Length: 344, dtype: bool

In [34]:
penguin['bill_length_mm'] <=40

0       True
1       True
2      False
3      False
4       True
       ...  
339    False
340    False
341    False
342    False
343    False
Name: bill_length_mm, Length: 344, dtype: bool

In [35]:
penguin['flipper_length_mm'] >= 200

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340     True
341    False
342     True
343    False
Name: flipper_length_mm, Length: 344, dtype: bool

<hr><h1>Cleaning the Data Frame</h1>

In [12]:
penguin.isnull()

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0      False   False           False          False              False   
1      False   False           False          False              False   
2      False   False           False          False              False   
3      False   False            True           True               True   
4      False   False           False          False              False   
..       ...     ...             ...            ...                ...   
339    False   False           False          False              False   
340    False   False           False          False              False   
341    False   False           False          False              False   
342    False   False           False          False              False   
343    False   False           False          False              False   

     body_mass_g    sex   year  
0          False  False  False  
1          False  False  False  
2          False  False  False  
3           True   True  False  
4          False  False  False  
..           ...    ...    ...  
339        False  False  False  
340        False  False  False  
341        False  False  False  
342        False  False  False  
343        False  False  False  

[344 rows x 8 columns]

In [13]:
penguin.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h1>What to do with all the NULL VALUES?</h1>

In [14]:
penguin['bill_length_mm']

0      39.1
1      39.5
2      40.3
3       NaN
4      36.7
       ... 
339    55.8
340    43.5
341    49.6
342    50.8
343    50.2
Name: bill_length_mm, Length: 344, dtype: float64

In [15]:
penguin['bill_length_mm'].mean()

np.float64(43.9219298245614)

In [16]:
penguin['sex'].mode()

0    male
Name: sex, dtype: object

<hr><h1>HOW TO FILLUP NULL VALUES</h1>

In [17]:
penguin['bill_length_mm'] = penguin['bill_length_mm'].fillna(penguin['bill_length_mm'].mean())

In [36]:
penguin['sex'] = penguin['sex'].fillna(penguin['sex'].mode()[0])

In [37]:
penguin.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        2
flipper_length_mm    2
body_mass_g          2
sex                  0
year                 0
dtype: int64

In [38]:
penguin['flipper_length_mm'] = penguin['flipper_length_mm'].fillna(penguin['flipper_length_mm'].mean())

In [39]:
penguin.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        2
flipper_length_mm    0
body_mass_g          2
sex                  0
year                 0
dtype: int64

In [19]:
penguin['sex'] = penguin['sex'].fillna(penguin['sex'].mode()[0])

In [20]:
penguin.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        2
flipper_length_mm    2
body_mass_g          2
sex                  0
year                 0
dtype: int64

<h1>DATA TYPES</h1>

In [40]:
penguin.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
count      344.000000     342.000000         344.000000   342.000000
mean        43.921930      17.151170         200.915205  4201.754386
std          5.443643       1.974793          14.020657   801.954536
min         32.100000      13.100000         172.000000  2700.000000
25%         39.275000      15.600000         190.000000  3550.000000
50%         44.250000      17.300000         197.000000  4050.000000
75%         48.500000      18.700000         213.000000  4750.000000
max         59.600000      21.500000         231.000000  6300.000000

<h1>CHANGING DATA TYPES</h1>

In [21]:
penguin['species'] = penguin['species'].astype('category')
penguin['year'] = penguin['year'].astype('category')

In [22]:
penguin.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    object  
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      342 non-null    float64 
 4   flipper_length_mm  342 non-null    float64 
 5   body_mass_g        342 non-null    float64 
 6   sex                344 non-null    object  
 7   year               344 non-null    category
dtypes: category(2), float64(4), object(2)
memory usage: 17.2+ KB


In [41]:
penguin.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
count      344.000000     342.000000         344.000000   342.000000
mean        43.921930      17.151170         200.915205  4201.754386
std          5.443643       1.974793          14.020657   801.954536
min         32.100000      13.100000         172.000000  2700.000000
25%         39.275000      15.600000         190.000000  3550.000000
50%         44.250000      17.300000         197.000000  4050.000000
75%         48.500000      18.700000         213.000000  4750.000000
max         59.600000      21.500000         231.000000  6300.000000

<h2>EXPLORING DATA ANALYSIS (EDA) </h1>

<h3>Which Specie of Penguin amongst the dataset is the heaviest?</h3>

In [23]:
penguin['body_mass_g'].mean()

np.float64(4201.754385964912)

In [24]:
adelie_df = penguin[penguin['species'] == "Adelie"]

In [25]:
penguin['body_mass_g'].mean()

np.float64(4201.754385964912)

In [26]:
penguin[penguin['species'] == "Adelie"]['body_mass_g'].mean()

np.float64(3700.662251655629)

<h3>1. How many penguins were observed on Torgesen island?</h3>

In [83]:
penguin.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    object  
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      342 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        342 non-null    float64 
 6   sex                344 non-null    object  
 7   year               344 non-null    category
dtypes: category(2), float64(4), object(2)
memory usage: 17.2+ KB


In [43]:
torgesen_data = penguin[penguin['island'] == 'Torgesen']

In [80]:
number_of_penguins = torgesen_data.shape[0]
number_of_penguins

0

<h4>INSIGHT: BASED ON THE RESULT ABOVE, IT SEEMS THAT THERE IS NO PENGUINS WERE OBSERVED ON TORGESEN ISLAND..</h4><hr>

<h3>2. How many Gentoo penguins are in the dataset?</h3><hr>

In [46]:
gentoo_count = penguin[penguin['species'] == 'Gentoo'].shape[0]

In [47]:
gentoo_count

124

<h4>ACCORDING FROM THE GIVEN DATA ABOVE, IT SHOWS THAT THERE ARE 124 PENGUINS ARE IN GENTOO</h4><hr>

<h3>3. What is the bill length of the first Adelie penguin listed?</h3><hr>

In [48]:
penguin

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000           18.7         181.000000   
1       Adelie  Torgersen        39.50000           17.4         186.000000   
2       Adelie  Torgersen        40.30000           18.0         195.000000   
3       Adelie  Torgersen        43.92193            NaN         200.915205   
4       Adelie  Torgersen        36.70000           19.3         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream        55.80000           19.8         207.000000   
340  Chinstrap      Dream        43.50000           18.1         202.000000   
341  Chinstrap      Dream        49.60000           18.2         193.000000   
342  Chinstrap      Dream        50.80000           19.0         210.000000   
343  Chinstrap      Dream        50.20000           18.7         198.000000   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN    male  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h4>ACCORDING FROM THE GIVEN DATA ABOVE, IT SHOW THAT THE BILL LENGTH OF THE FIRST ADELIE PENGUIN THAT LISTED FROM THE DATASET IS 39.10000</h4><hr>

<h3>4. Are there any missing values for the bill length?</h3><hr>

In [49]:
penguin['bill_length_mm'].isnull().sum()

np.int64(0)

<h4>INSIGHT: ACCORDING FROM THE GIVEN DATA ABOVE, IT SHOWS THAT THERE ARE NO ANY MISSING VALUE FOR THE BILL LENGTH OF EACH SPECIES</h4><hr>

<h3>5. For the Adelie species, what is the average body mass of male penguins compared to female penguins?</h3><hr>

In [50]:
penguin.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    object  
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      342 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        342 non-null    float64 
 6   sex                344 non-null    object  
 7   year               344 non-null    category
dtypes: category(2), float64(4), object(2)
memory usage: 17.2+ KB


In [51]:
adelie_df = penguin[penguin['species'] == 'Adelie']

In [52]:
male_mass = adelie_df[adelie_df['sex'] == 'male']['body_mass_g'].mean()

In [53]:
female_mass = adelie_df[adelie_df['sex'] == 'female']['body_mass_g'].mean()

In [54]:
male_mass, female_mass

(np.float64(4011.2179487179487), np.float64(3368.8356164383563))

<h4>INSIGHT: BASED FROM THE GIVEN DATA ABOVE, IT SHOWS THAT THE AVERAGE BODY MASS OF MALE PENGUINS ARE 4011.2179487179487 WHILE FEMALE
PENGUINS ARE 3368.8356164383563 THAT COVERS ONLY ON THE ADELIE ISLAND.</h4><hr>

<h3>6. Which island has the highest number of observed penguins?</h3><hr>

In [55]:
penguin['island'].value_counts()

island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

<h4>INSIGHT: BASED FROM THE GIVEN DATA ABOVE, IT SHOWS THAT THE HIGHEST NUMBER OF OBSERVED PENGUINS WHERE AT THE BISCOE ISLAND </h4><hr>

<h3>7. Which species of penguin generally has the longest flippers?</h3><hr>

In [56]:
penguin.groupby('species')['flipper_length_mm'].mean()

species
Adelie       190.025758
Chinstrap    195.823529
Gentoo       217.055768
Name: flipper_length_mm, dtype: float64

<h4>INSIGHT: BASED FROM THE GIVEN DATA ABOVE, IT INDICATE THAT THE LONGEST FLIPPERS AMONGST THEM IS FROM THE GENTOO ISLAND THAT HAS AN 217.186992 </h4><hr>

<h3>8. What is the average flipper length of penguins observed on Torgersen Island compared to Biscoe Islands?</h3><hr>

In [57]:
torgersen_avg = penguin[penguin['island'] == 'Torgersen']['flipper_length_mm'].mean()

In [58]:
biscoe_avg = penguin[penguin['island'] == 'Biscoe']['flipper_length_mm'].mean()

In [59]:
torgersen_avg, biscoe_avg

(np.float64(191.38298470535312), np.float64(209.65425717070454))

<h4>INSIGHT: BASED FROM THE GIVEN DATA ABOVE, IT SHOWS THAT THE AVERAGE FLIPPER LENGTH OF PENGUINS OBSERVED ON TORGENSEN ISLAND ARE 191.19607843137254
COMPARED TO BISCOE ISLANDS ARE 209.70658682634732 </h4><hr>

<h3>9. What is the bill depth of the last Chinstrap penguin listed?</h3><hr>

In [60]:
chinstrap_penguins = penguin[penguin['species'] == 'Chinstrap']

In [61]:
chinstrap_penguins.iloc[-1]['bill_depth_mm']

np.float64(18.7)

<h4>INSIGHT: ACCORDING FROM THE GIVEN DATA ABOVE, IT SHOWS THAT THE BILL LENGTH OF THE LAST CHINSTRAP THAT WERE LISTED FROM THE DATA ARE 18.7</h4><hr>

<h3> 10. Are there any missing values for flipper length?</h3><hr>

In [62]:
penguin['flipper_length_mm'].isnull().sum()

np.int64(0)

<h4>INSIGHT: BASED FROM THE GIVEN DATA ABOVE, IT SHOWS THAT THERE ARE 2 MISSING VALUES FOR FLIPPER LENGTH </h4><hr>

<h3>11. Are there any missing for body mass?</h3><hr>

In [63]:
penguin['flipper_length_mm'].isnull().sum()

np.int64(0)

<h4>INSIGHT: ACCORDING FROM THE GIVEN DATA ABOVE, IT SHOWS THAT THERE ARE 2 MISSING FOR BODY MASS FROM THE DATASET</h4><hr>

<h3>12.How many records are there for each penguin species?</h3><hr>

In [64]:
penguin['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

<h4>INSIGHT: ACCORDING FROM THE GIVEN DATA ABOVE, IT INDICATE THE ADELIE HAS THE MOST FREQUENTLY OBSERVED, FOLLOWED BY GENTOO, WHILE CHINSTRAP 
HAS THE LEAST NUMBER OF OBSERVED SPECIES.</h4><hr>

<h3>13. What is the body mass of the last Chinstrap penguin listed from Dream island in 2009?</h3><hr>

In [65]:
chinstrap_dream_2009 = penguin[
    (penguin['species'] == 'Chinstrap') &
    (penguin['island'] == 'Dream') &
    (penguin['year'] == 2009)
]

In [66]:
chinstrap_dream_2009.iloc[-1]['body_mass_g']

np.float64(3775.0)

<h4>INSIGHT: BASED FROM THE GIVEN DATA ABOVE, IT SHOWS THAT THE BODY MASS OF THE LAST CHINSTRAP PENGUIN THAT LISTED FROM DREAM ISLAND IN 2009 IS 3775.0  </h4><hr>

<h1>DATA AGGREGATION</h1>

In [67]:
#species, island, sex

sp_is_x = penguin.groupby(['species', 'island', 'sex']).size()

In [68]:
sp_is_x 

species    island     sex   
Adelie     Biscoe     female    22
                      male      22
           Dream      female    27
                      male      29
           Torgersen  female    24
                      male      28
Chinstrap  Biscoe     female     0
                      male       0
           Dream      female    34
                      male      34
           Torgersen  female     0
                      male       0
Gentoo     Biscoe     female    58
                      male      66
           Dream      female     0
                      male       0
           Torgersen  female     0
                      male       0
dtype: int64

<h1>DETERMINING CORRELATION USING PANDAS PEARSON R</h1>
<hr>
<h3>Is there a correlation between body mass and flipper?</h3>

In [69]:
mass_flip_cor = penguin ['flipper_length_mm'].corr(penguin['body_mass_g'])

In [70]:
mass_flip_cor

np.float64(0.8712017673060113)

<h1>Categorical to Numerical Correlation</h1>
<h2>Using One Way Anova</h2>
<h>Library : Scientific Python ( SCIPY )</h2>

In [71]:
pip install scipy

Note: you may need to restart the kernel to use updated packages.


In [72]:
from scipy.stats import f_oneway

In [73]:
male_df = penguin[penguin['sex'] == 'male']

In [74]:
male_df['body_mass_g']

0      3750.0
3         NaN
5      3650.0
7      4675.0
8      3475.0
        ...  
334    3800.0
336    3950.0
339    4000.0
341    3775.0
342    4100.0
Name: body_mass_g, Length: 179, dtype: float64

In [75]:
penguin[penguin['sex'] == 'male'] ['body_mass_g']

0      3750.0
3         NaN
5      3650.0
7      4675.0
8      3475.0
        ...  
334    3800.0
336    3950.0
339    4000.0
341    3775.0
342    4100.0
Name: body_mass_g, Length: 179, dtype: float64

In [76]:
penguin[penguin['species'] == 'Adelie'] ['body_mass_g']

0      3750.0
1      3800.0
2      3250.0
3         NaN
4      3450.0
        ...  
147    3475.0
148    3450.0
149    3750.0
150    3700.0
151    4000.0
Name: body_mass_g, Length: 152, dtype: float64

In [77]:
penguin[penguin['species'] == 'Gentoo'] ['body_mass_g']

152    4500.0
153    5700.0
154    4450.0
155    5700.0
156    5400.0
        ...  
271       NaN
272    4850.0
273    5750.0
274    5200.0
275    5400.0
Name: body_mass_g, Length: 124, dtype: float64

In [78]:
penguin[penguin['species'] == 'Chinstrap'] ['body_mass_g']

276    3500.0
277    3900.0
278    3650.0
279    3525.0
280    3725.0
        ...  
339    4000.0
340    3400.0
341    3775.0
342    4100.0
343    3775.0
Name: body_mass_g, Length: 68, dtype: float64

In [79]:
f_oneway(penguin[penguin['species'] == 'Adelie'] ['body_mass_g'], penguin[penguin['species'] == 'Gentoo'] ['body_mass_g'], 
penguin[penguin['species'] == 'Chinstrap'] ['body_mass_g'])

F_onewayResult(statistic=np.float64(nan), pvalue=np.float64(nan))